### PR TITLE
Add detail view

### LINKED ISSUE #23: Create the time entry detail view
Add the time entry detail view and bind it with the list view.

### PR DESCRIPTION
**Pull request by Odrakir**
Closes #23 

This is not using the designs in Zeplin because they are not final, but it has all the pieces.

Pay special attention to the list of tags that uses a custom component so the list wraps at the end of the line. 

And also to the running timer when the entry is running.

### CODE REVIEW

File: TogglTrack/Core/State/AppState.swift [hunk 1]
```diff
@@ -75,22 +75,9 @@ public let timeEntryModelSelector = memoize{ (state: TimelineState) -> [TimeEntr
 
 extension TimelineState
 {
-    public var timeEntryModels: [TimeEntryModel]
```
[the remark this answers is not in the document]
Comment (Odrakir) [reply TogglTrack/Core/State/AppState.swift]: It's on develop, so I'm not sure, but whatever happened there, it needs to go.

File: TogglTrack/Views/WrappedListCell.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+//  WrappedListCell.swift
+//  TogglWatch WatchKit Extension
+//
+//  Created by Ricardo Sánchez Sotres on 27/11/2019.
+//  Copyright © 2019 Toggl. All rights reserved.
+//
+
+import SwiftUI
+
+struct WrappedListSizeKey: PreferenceKey {
```
[the remark this answers is not in the document]
Comment (Odrakir) [reply TogglTrack/Views/WrappedListCell.swift]: No idea what's going on here. I've searched online with no result. Lets leave it for now.

**Comment by Odrakir** [comment]
No idea what's going on with the `WrappedListSizeKey`. I've searched online with no result. Lets leave it for now.

File: TogglTrack/Model/TimeEntryModel.swift [hunk 3]
```diff
@@ -73,7 +73,7 @@ public struct TimeEntryModel
         self.id = timeEntry.id
         self.description = timeEntry.description
         self.start = timeEntry.start
-        self.duration = timeEntry.duration
+        self.duration = timeEntry.duration >= 0 ? timeEntry.duration : nil
```
Comment (sofokli1) [inline TogglTrack/Model/TimeEntryModel.swift]: do we get a negative value from the server for timeEntry? if yes, then ok. if no, then `timeEntry.duration >= 0` doesn't make sense.

File: TogglTrack/Model/TimeEntry.swift [hunk 4]
```diff
@@ -13,7 +13,7 @@ public struct TimeEntry: Codable, Equatable, Identifiable
     public var id: Int
     public var description: String
     public var start: Date
-    public var duration: Double?
```
Comment (sofokli1) [inline TogglTrack/Model/TimeEntry.swift]: if you remove the optional, how do we knwo if it is running or not?

File: TogglTrack/Model/TimeEntryModel.swift (hunk 3 above)
Comment (Odrakir) [reply TogglTrack/Model/TimeEntryModel.swift]: For the running TE the duration is negative, yes. And duration is never nil, that's why I did this change.

File: TogglTrack/Model/TimeEntry.swift (hunk 4 above)
Comment (Odrakir) [reply TogglTrack/Model/TimeEntry.swift]: See above. Running TimeEntry has a negative duration, not nil. I do turn it to nil in the case of TimeEntryModel

File: TogglWatch WatchKit Extension/Views/TimeEntry/TimeEntryDetailView.swift [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+struct DurationView: View
+{
+    var timeEntry: TimeEntryModel
+    @State var timer: ObservableTimer = ObservableTimer()
+    @State var now = Date()
+
+    var body: some View
+    {
+        Group {
+            if self.timeEntry.durationString != nil {
```
[the remark this answers is not in the document]
Comment (Odrakir) [reply TogglWatch WatchKit Extension/Views/TimeEntry/TimeEntryDetailView.swift]: As far as I know, no, it doesn't.

**Pull request merged**